### PR TITLE
upgraded golangci-lint to 1.54.2

### PR DESCRIPTION
**Pull request by Nikhil-Ladha**
Upgraded golangci-lint to 1.54.2 and resolved the errors reported.

### CODE REVIEW

File: controllers/storagecluster/generate.go [hunk 1]
```diff
@@ -111,10 +109,6 @@ func generateNameForCephNetworkFilesystemSC(initData *ocsv1.StorageCluster) stri
 	return fmt.Sprintf("%s-ceph-nfs", initData.Name)
 }
 
-func generateNameForNFSCSIProvisioner(initData *ocsv1.StorageCluster) string {
```
Comment (iamniting) [inline controllers/storagecluster/generate.go:114]: Why are we removing this func. I see this was being used.

File: controllers/storageconsumer/storageconsumer_controller.go [hunk 2]
```diff
@@ -71,7 +71,7 @@ type StorageConsumerReconciler struct {
 // Note:
 // The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+func (r *StorageConsumerReconciler) Reconcile(_ context.Context, request reconcile.Request) (reconcile.Result, error) {
```
Comment (iamniting) [inline controllers/storageconsumer/storageconsumer_controller.go:74]: Lets assign this ctx value to ctx in StorageConsumerReconciler `r.ctx = ctx`

**Review by iamniting — CHANGES_REQUESTED** [review]

**Comment by iamniting** [comment]
@nb-ohad pls take a look at the server code.

File: controllers/storagecluster/generate.go (hunk 1 above)
Comment (Nikhil-Ladha) [reply controllers/storagecluster/generate.go:114]: IT was only used in one place and that too not consuming the parameter passed to it. So, removed it to directly use the return value of `fmt.Sprintf`.
We are doing a similar thing for a function just above it as well.
Ref: https://github.com/red-hat-storage/ocs-operator/blob/fbfe2d9ea0256d9e38ee57ca911c74169cf55f6d/controllers/storagecluster/storageclasses.go#L377

File: controllers/storageconsumer/storageconsumer_controller.go (hunk 2 above)
Comment (Nikhil-Ladha) [reply controllers/storageconsumer/storageconsumer_controller.go:74]: Ack

**Review by umangachapagain — COMMENTED** [review]
v1.55.2 is already available. Any reason to not move to that?
Locally other versions did not work for me with go1.21.4.

**Comment by Nikhil-Ladha** [comment]
> v1.55.2 is already available. Any reason to not move to that?

No reason as such, 1.54 was the chronologically first version that supported go1.21 so updated to that.

> Locally other versions did not work for me with go1.21.4.

Interesting, in CI also we are using go1.21.4 and everything seems to work fine 🤔 
Ref: https://github.com/red-hat-storage/ocs-operator/actions/runs/7194272521/job/19594462514?pr=2319

**Review by iamniting — APPROVED** [review]

**Pull request merged**